fontra2fontir/src/fontra.rs

```diff
@@ -0,0 +1,281 @@
+//! Parse .fontra json filesets into structs
+//!
+//! See <https://github.com/googlefonts/fontra/blob/main/src/fontra/core/classes.py>
+
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use fontdrasil::types::GlyphName;
+use fontir::error::Error;
+use percent_encoding::{utf8_percent_encode, NON_ALPHANUMERIC};
+use serde::Deserialize;
+use write_fonts::types::Tag;
+
+pub(crate) fn glyph_file(glyph_dir: &Path, glyph: GlyphName) -> PathBuf {
+    // TODO: this probably over-encodes. Should be sufficient to get started.
+    let filename = utf8_percent_encode(glyph.as_str(), NON_ALPHANUMERIC);
+    glyph_dir.join(filename.to_string() + ".json")
+}
+
+fn from_file<T>(p: &Path) -> Result<T, Error>
+where
+    for<'a> T: Deserialize<'a>,
+{
+    let raw = fs::read_to_string(p).map_err(Error::IoError)?;
+    serde_json::from_str(&raw).map_err(|e| Error::ParseError(p.to_path_buf(), format!("{e}")))
+}
+
+/// serde type used to load font-data.json
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct FontraFontData {
+    #[serde(rename = "unitsPerEm")]
+    pub(crate) units_per_em: u16,
+    #[serde(default)]
+    pub(crate) axes: Vec<FontraAxis>,
+}
+
+impl FontraFontData {
+    pub(crate) fn from_file(p: &Path) -> Result<Self, Error> {
+        from_file(p)
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub(crate) struct FontraAxis {
+    pub(crate) name: String,
+    pub(crate) tag: Tag,
+    #[serde(default)]
+    pub(crate) hidden: bool,
+    #[serde(rename = "minValue")]
+    pub(crate) min_value: f64,
+    #[serde(rename = "defaultValue")]
+    pub(crate) default_value: f64,
+    #[serde(rename = "maxValue")]
+    pub(crate) max_value: f64,
+    #[serde(default)]
+    pub(crate) mapping: Vec<[f64; 2]>,
+}
+
+/// serde type used to load .fontra/glyphs/namelike.json files
+///
+/// <https://github.com/googlefonts/fontra/blob/a4edd06837118e583804fd963c22ed806a315b04/src/fontra/core/classes.py#L104-L116>
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)] // TEMPORARY
```

Comment:
I have no real preference from a review perspective other than the noise, but I think it's easier to just add `#![allow(dead_code)]` at the top of the module when pushing incremental PRs.

Reply:
The noise is annoying indeed. I actually prefer to just prefix fields with _ and not so annotate at all.